llama-index-core/llama_index/core/indices/struct_store/sql_query.py

```diff
@@ -422,7 +422,7 @@ def _query(self, query_bundle: QueryBundle) -> Response:
                 query=query_bundle.query_str,
                 nodes=retrieved_nodes,
             )
-            cast(Dict, response.metadata).update(metadata)
+            # cast(Dict, response.metadata).update(metadata)
```

Comment:
Why comment this out? Doesn't this mean metadata doesn't propagate from the retriever?

Reply:
Pretty sure this is a breaking change? Metadata should be on the response object no?

Reply:
Seems that currently the behaviour of nl2sql query engines derived from `BaseSQLTableQueryEngine` is inconsistent with other types of database query engine (e.g. `KnowledgeGraphQueryEngine`). 

`KnowledgeGraphQueryEngine`, like the vector index query engines,  propagate the metadata through the retrived nodes, which results in putting the response metadata (e.g. `cypher`, `graph_store_response`) under the node id key in the dictionary (see the exepcted behaviour in the PR description for the difference).

On the other hand `BaseSQLTableQueryEngine` propagate metadata through `retrieve_with_metadata` and does not set the metadata in the node. This problem cause problem when parsing metadata when using it to compose complex query engine/retriever like `RouterRetriever`.

If you think we should avoid having this change, i can update the PR to keep this line, it will still fix the problem of parsing metadata when using it for `RouterRetriever`.

Reply:
`KnowledgeGraphQueryEngine` is basically dead, especially given the recent property graph index.  I wouldn't worry about conforming to whatever it does :)

Reply:
Thank you! I have updated the PR to keep the propagated metadata.

Reply:
Thanks!